frontend/src/store/search/reducer.ts

```diff
@@ -0,0 +1,65 @@
+import { createReducer } from '@reduxjs/toolkit';
+
+import { TypeFilterId, DateFilterId, DurationFilterId, SortByFilterId, SearchQueryParam } from './models';
+import {
+  setSearchText,
+  setActiveTypeFilterId,
+  setActiveDateFilterId,
+  setActiveDurationFilterId,
+  setActiveSortByFilterId,
+  clearActiveFilterIds,
+} from './actions';
+
+type State = {
+  searchText: string;
+  searchUrlParams: string;
```

Comment:
Not sure we need this state in the redux... You need to have one source of true and it should be `activeFilterId` field. Let me know if you have problems on it I will help you